Lib/email/_header_value_parser.py

```diff
@@ -2849,8 +2852,14 @@ def _refold_parse_tree(parse_tree, *, policy):
                 last_ew = _fold_as_ew(tstr, lines, maxlen, last_ew,
                                       part.ew_combine_allowed, charset)
                 last_charset = charset
-            want_encoding = False
-            continue
+                want_encoding = False
+                continue
+            else:
+                # It's a terminal which should be kept non-encoded
+                # (e.g. a ListSeparator).
+                last_ew = None
+                want_encoding = False
+                pass
```

Comment:
We can remove that pass I think :)
```suggestion
```

Reply:
Indeed, this is not needed.

I experimented with variants that do not reset `last_ew` and `want_encoding`, but have not found examples in which this would make difference. Therefore, they are reset just in case, without confirming the need.